Tests/GeoLib/TestDuplicateGeometry.cpp

```diff
@@ -0,0 +1,153 @@
+    int box_size (rand());
+    double half_box_size(box_size/2);
+    std::unique_ptr<std::vector<GeoLib::Point*>> pnts (new std::vector<GeoLib::Point*>);
+    pnts->reserve(n_pnts);
+    for (int k(0); k<n_pnts; k++) {
+        pnts->push_back(new GeoLib::Point(
+            rand() % box_size - half_box_size,
+            rand() % box_size - half_box_size,
+            rand() % box_size - half_box_size));
+    }
+    geo.addPointVec(std::move(pnts), input_name);
+    std::string output ("output_geometry");
+
+    // duplicate points
+    {
+        GeoLib::DuplicateGeometry dup (geo, input_name, output);
+
+        std::vector<GeoLib::Point*> const*const pnts (geo.getPointVec(input_name));
+        std::vector<GeoLib::Point*> const*const new_pnts (geo.getPointVec(output));
+        std::vector<GeoLib::Point*>& mod_pnts (dup.getPointVectorCopy());
+        ASSERT_EQ(n_pnts, new_pnts->size());
+        ASSERT_EQ(n_pnts, mod_pnts.size());
+
+        for (std::size_t i=0; i<n_pnts; ++i)
+        {
+            ASSERT_EQ((*(*pnts)[i])[0], (*(*new_pnts)[i])[0]);
+            ASSERT_EQ((*(*pnts)[i])[1], (*(*new_pnts)[i])[1]);
+            ASSERT_EQ((*(*pnts)[i])[2], (*(*new_pnts)[i])[2]);
+        }
+        mod_pnts.push_back(new GeoLib::Point(0,0,0));
+        ASSERT_EQ(mod_pnts.size(), pnts->size() + 1);
+        ASSERT_EQ(mod_pnts.size(), new_pnts->size());
+    }
+
+    // add polylines
+    std::size_t n_plys (rand() % 10 + 1);
+    std::unique_ptr<std::vector<GeoLib::Polyline*>> plys (new std::vector<GeoLib::Polyline*>);
+    for (std::size_t i=0; i<n_plys; ++i)
+    {
+        int n_ply_pnts (rand() % 100 + 2);
+        GeoLib::Polyline* line = new GeoLib::Polyline(*geo.getPointVec(input_name));
+        for (std::size_t j=0; j<n_ply_pnts; ++j)
+            line->addPoint(rand() % n_pnts);
+        plys->push_back(line);
+    }
+    geo.addPolylineVec(std::move(plys), input_name);
+
+    // duplicate polylines
+    {
+        GeoLib::DuplicateGeometry dup (geo, input_name, output);
+        std::string const& output2 = dup.getFinalizedOutputName();
+        ASSERT_FALSE(output == output2);
+
+        std::vector<GeoLib::Polyline*> const*const plys (geo.getPolylineVec(input_name));
+        std::vector<GeoLib::Polyline*> const*const new_plys (geo.getPolylineVec(output2));
+        ASSERT_EQ(n_plys, new_plys->size());
+
+        for (std::size_t i=0; i<n_plys; ++i)
+        {
+            std::size_t const n_ply_pnts ((*new_plys)[i]->getNumberOfPoints());
+            ASSERT_EQ(n_ply_pnts, (*plys)[i]->getNumberOfPoints());
+            ASSERT_EQ((*new_plys)[i]->getNumberOfSegments(), (*plys)[i]->getNumberOfSegments());
+            for (std::size_t j=0; j<n_ply_pnts; ++j)
+                ASSERT_EQ((*plys)[i]->getPointID(j), (*new_plys)[i]->getPointID(j));
+        }
+
+        std::vector<GeoLib::Polyline*>& mod_plys (dup.getPolylineVectorCopy());
+        mod_plys.push_back(new GeoLib::Polyline(*(*new_plys)[0]));
```

Comment:
adding a polyline to the copied vector to test if it's possible and if the correct result is returned.